contracts/multisig/src/signing.rs

```diff
@@ -29,47 +29,64 @@ impl SigningSession {
         }
     }
 
+    /// returns true if session state was changed
     pub fn recalculate_session_state(
         &mut self,
         signatures: &HashMap<String, Signature>,
         snapshot: &Snapshot,
         block_height: u64,
-    ) {
+    ) -> bool {
```

Comment:
I wonder if it would be cleaner to just return an event here. I think we are only using this return value to determine whether or not to add an event into the response later on, so might be cleaner to just return the event itself here.

Reply:
I'm not sure I like the idea of mixing events with logic, I'll prefer to have them separated the same way we are trying to keep storage separated from logic

Reply:
That's true. I just don't like this `bool` return value here. I feel like another option is to just return the new state, and compare that to the old state where the function is being called. Or don't even return the new state, just mutate the state, and then compare old state to new state where it's being called. Having the bool return value here just feels like how this function is being used is "bleeding" into the implementation here. But up to you, I don't feel super strongly about it.